data_and_models/pipelines/ner/clean.py

```diff
@@ -0,0 +1,121 @@
+# Blue Brain Search is a text mining toolbox focused on scientific use cases.
+#
+# Copyright (C) 2020  Blue Brain Project, EPFL.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+"""
+This script cleans annotations exported with Prodigy as .jsonl files.
+
+Run 'analyze.py' before to identify relevant cleaning rules.
+
+Usage: clean.py [OPTIONS] INPUT_PATH KEEP_LABEL [RENAME_INTO]
+
+First, this script keeps only valid texts from the annotations (see next paragraph).
+Second, this script normalizes entity labels (use upper case for labels in lower case).
+Third, this script keeps only the given entity label and renames it if necessary.
+
+Valid texts are texts which have been accepted, have or not spans (i.e. entities), and
+have a unique 'input_hash'.
+
+The output is one file. The file is put in the same directory as the input file. It is
+named following the pattern 'annotations_<entity_label>.jsonl'.
+"""
+
+from collections import defaultdict
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import srsly
+import typer
+
+
+def is_valid(example: Dict[str, Any], duplicated_hashes: set) -> bool:
+    accepted = example["answer"] == "accept"
+    unique_hash = example["_input_hash"] not in duplicated_hashes
+    return all((accepted, unique_hash))
+
+
+def main(
+    input_path: Path = typer.Argument(..., exists=True, dir_okay=False),
+    keep_label: str = typer.Argument(...),
+    rename_into: Optional[str] = typer.Argument(None),
+):
+    keep_label = keep_label.upper()
+    rename_into = None if rename_into is None else rename_into.upper()
+
+    print("Read annotations...")
+    corpus = list(srsly.read_jsonl(input_path))
```

Comment:
spaCy people created and use srsly. So, I prefer to use it than to reinvent the wheel.

I think it was not moved by #345 to extra. But it could.

Reply:
Regarding the leftovers from #345, I have created a dedicated issue to avoid out-of-scope work in this PR:
https://github.com/BlueBrain/Search/issues/368